src/Spatial.Tests/Euclidean/Circle2DTests.cs

```diff
@@ -58,5 +59,51 @@ public void CircleFromThreePointsArgumentException()
 
             Assert.Throws<ArgumentException>(() => { Circle2D.FromPoints(p1, p2, p3); });
         }
+
+        [TestCase("0,0", 1, "-10,+10", "+10,+10", 0)]
+        [TestCase("0,0", 1, "-10,+1", "+10,+1", 1)]
+        [TestCase("0,0", 1, "-10,0", "+10,0", 2)]
+        [TestCase("0,0", 1, "-10,-1", "+10,-1", 1 )]
+        [TestCase("0,0", 1, "-10,-10", "+10,-10", 0)]
+        public void CircleIntersectWithLine2D_NumberOfIntersections(string sc, double radius, string sps, string spe, int expectedNumberOfIntersections)
+        {
+            var circle = new Circle2D(Point2D.Parse(sc), radius);
+            var line = new Line2D(Point2D.Parse(sps), Point2D.Parse(spe));
+
+            var actual = circle.IntersectWith(line);
+
+            Assert.That(actual.Count(), Is.EqualTo(expectedNumberOfIntersections));
+            //TODO: the intersection should be on the circle
+            Assert.That(actual.All(p => Math.Abs(circle.Center.DistanceTo(p) - circle.Radius) < 1e-6), Is.EqualTo(true), "distance between center and intersection");
+            //TODO: the intersection should be on the line
+        }
+
+        //segment contains the all intersections(same to the cases of circle and line)
+        [TestCase("0,0", 1, "-10,+10", "+10,+10", 0)]
```

Comment:
Same comment here as in the test above (line always parallel to X)

Reply:
Thanks for the comment.
I will add tests(line parallel to Y for example, and the general case).

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/3d3748bca4569c6d5a7aed16c1c3957a1da9e8ff